vault/identity_store_util.go

```diff
@@ -171,7 +171,16 @@ func (i *IdentityStore) loadGroups(ctx context.Context) error {
 			}
 
 			err = i.UpsertGroupInTxn(ctx, txn, group, persist)
-			if err != nil {
+
+			if errors.Is(err, logical.ErrReadOnly) {
+				// This is an imperfect solution to unblock customers who are running into
+				// a readonly error during a DR failover (jira #28191). More specifically, there
+				// are duplicate aliases in storage that are merged during loadEntities. Vault
+				// attempts to remove these deleted entities from the group, but fails in the case
+				// where the node is a PR secondary because the RPC client is not yet initialized
+				// and the storage is read-only.
+				i.logger.Warn("received a read only error while trying to upsert group to storage")
```

Comment:
Maybe I missed this, but when will it be updated, then? Do we need to update it in storage at all?

Reply:
No in this particular case (when we get the ErrReadOnly) we are skipping the step to persist it to storage.

Reply:
Sorry, I might be missing something, but how and when will all the nodes update memdb correctly? I remember reading something about this code being here to fix another issue.

Reply:
Don't apologize :) there are so many moving pieces and I appreciate the questions, I may be missing something myself. The [UpsertGroupInTxn] also updates the group in memdb [here](https://github.com/hashicorp/vault/blob/main/vault/identity_store_util.go#L1835). So in the case when we see the `ErrReadOnly` we are not rolling back the entire transaction, so even if the storage update fails the memdb update will not.

Reply:
And when will the storage pick up the changes? I think what I mean is: when will storage and memdb synchronize? After the WALs get to the cluster replica/node replica? 

Are we aware/ok with the race conditions this might cause? For example, issuing an update/delete to the group right after the cluster has initialized and if the WALs aren't up to date?